test/distributed/test_pg_wrapper.py

```diff
@@ -65,59 +88,90 @@ def _test_collectives_op_mismatch(self, wrapper_pg, use_cuda=False):
             w.wait()
 
         # Simulate mismatch: allreduce vs reduce.
-        with self.assertRaisesRegex(
-            RuntimeError, "Mismatch between collective operation types"
-        ):
+        # Error including info about inconsistent collective, rank, tensor
+        # shape, device, and dtype should be raised.
+        with self.assertRaisesRegex(RuntimeError, "..") as cm:
             if self.rank == 0:
                 wrapper_pg.allreduce([tensor])
             else:
                 wrapper_pg.reduce([tensor])
+            self._validate_error(
```

Comment:
`_validate_error` calls should be indented with the `with` statements. Basically `with self.assert*` and `_validate_error` are two separate assertions. They are not cascaded.

Reply:
Thanks for catching this!